core/framework/condition_collection.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import "github.com/kiegroup/kogito-cloud-operator/api"
+
+const maxBufferCondition = 5
+
+// ConditionCollection ...
+type ConditionCollection interface {
+	AddCondition(condition api.ConditionInterface) []api.ConditionInterface
+}
+
+type conditionCollection struct {
+	conditions []api.ConditionInterface
+}
+
+// NewConditionCollection ...
+func NewConditionCollection(conditions []api.ConditionInterface) ConditionCollection {
+	return &conditionCollection{
+		conditions: conditions,
+	}
+}
+
+func (c *conditionCollection) AddCondition(newCondition api.ConditionInterface) []api.ConditionInterface {
+	for i, condition := range c.conditions {
+		if condition != nil && condition.GetType() == newCondition.GetType() {
+			c.removeOldCondition(i)
+		}
+	}
+	c.pushCondition(newCondition)
+	return c.conditions
+}
+
+func (c *conditionCollection) removeOldCondition(index int) {
+	c.conditions = append(c.conditions[:index], c.conditions[index+1:]...)
+}
+
+func (c *conditionCollection) pushCondition(newCondition api.ConditionInterface) {
+	if len(c.conditions) >= maxBufferCondition {
```

Comment:
What is the purpose of the bugger here, when every condition type is available just once?

Reply:
`maxBufferLimit` was already there in previous implementation that why I keep it.
yes you are right we can remove it now.

Reply:
done